docs/syncer.md

```diff
@@ -11,77 +11,79 @@ users have to install a special component named [syncer](https://github.com/kcp-
+## For syncer development
 
-```sh
-$ kubectl get secret
-NAME                  TYPE                                  DATA   AGE
-default-token-nkpxj   kubernetes.io/service-account-token   3      29m
-syncer-token-2mtgf    kubernetes.io/service-account-token   3      29m
+Alternately, create a `kind` cluster with a local registry to simplify syncer development by executing the
+following script:
 
-```
+https://raw.githubusercontent.com/kubernetes-sigs/kind/main/site/static/examples/kind-with-registry.sh
+
+### Building the syncer image
 
-Obtain the cluster certificate and the token associated to the service account:
+Install `ko`:
 
 ```sh
-$ KCP_SYNCER_TOKEN=$(kubectl get secret/syncer-token-2mtgf -o jsonpath='{.data.token}')
-$ KCP_SYNCER_CACRT=$(kubectl get configmap/kube-root-ca.crt -o jsonpath='{.data.ca\.crt}'| base64 -w 0)
+go install github.com/google/ko@latest
 ```
 
+Build image and push to the local registry integrated with `kind`:
 
-4. Complete the [syncer manifest](../manifest/syncer.yaml) with the information obtained in previous steps:
+```sh
+KO_DOCKER_REPO=localhost:5001 ko publish ./cmd/syncer -t <your tag>
+```
+
+By default `ko` will build for `amd64`. To build for `arm64` (e.g. apple silicon), specify
+`--platform=linux/arm64`.
 
-- KCP_SYNCER_CACRT
-- KCP_SYNCER_TOKEN
-- KCP_SERVER_URL
-- KCP_LOGICAL_CLUSTER_NAME
-- WORKLOAD_CLUSTER_NAME (defined by the user to identify the workload cluster)
+To use the image pushed to the local registry, supply `--image=<image tag>` to the
+`enable-syncer` plugin command, where `<image tag>` is from the output of `ko publish`.
 
-5.  Apply the manifest to the p-cluster
+1. Apply the manifest to the p-cluster
 
 ```sh
-$ kubect --kubeconfig pcluster.kubeconfig apply -f syncer.yaml
-namespace/kcp-system created
-serviceaccount/syncer created
-clusterrole.rbac.authorization.k8s.io/syncer created
-clusterrolebinding.rbac.authorization.k8s.io/syncer created
-secret/syncer-kcp-sa created
-configmap/syncer-kcp-config created
-deployment.apps/syncer created
+$ kubectl apply -f syncer.yaml
+namespace/kcpsync25e6e3ce5be10b16411448aec95b6b6d695a1daa5120732019531d8d created
+serviceaccount/kcp-syncer created
+clusterrole.rbac.authorization.k8s.io/kcpsync25e6e3ce5be10b16411448aec95b6b6d695a1daa5120732019531d8d created
+clusterrolebinding.rbac.authorization.k8s.io/kcpsync25e6e3ce5be10b16411448aec95b6b6d695a1daa5120732019531d8d created
+secret/kcp-syncer-config created
+deployment.apps/kcp-syncer created
 ```
 
-and it will create a deployment with the `syncer`:
+and it will create a `kcp-syncer` deployment:
 
 ```sh
-$ kubectl -n kcp-system get deployments
+$ kubectl -n kcpsync25e6e3ce5be10b16411448aec95b6b6d695a1daa5120732019531d8d get deployments
 NAME     READY   UP-TO-DATE   AVAILABLE   AGE
-syncer   1/1     1            1           13m
-```
\ No newline at end of file
+kcp-syncer   1/1     1            1           13m
+```
+
+1. Wait for the kcp workload cluster to go ready.
+
+TODO(marun)
```

Comment:
Is this for this PR or a follow up?

Reply:
Follow-up. I typically think of FIXME as something to finish in the current PR and TODO as 'eventually'.